README.md

```diff
@@ -78,6 +80,39 @@ class StaticController < ApplicationController
 end
 ```
 
+`add_tracer` can add a custom span to the specified instance or class method that is already defined. `add_tracer` allow to set custom span name, span kind and additional attributes in hash format
+
+```ruby
+add_tracer :function_name, 'custom_span_name', { attributes: { 'any' => 'attributes' }, kind: :span_kind }
+```
+
+For example, if you want to instrument class function or instance function `create_session` inside application controller
+
+```ruby
+class SessionsController < ApplicationController
+  include SolarWindsAPM::API::Tracer
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    create_session(user)
+  end
+
+  # instrument class function create_session
+  def create_session(user)
+  end
+  add_tracer :create_session, 'custom_name', { attributes: { 'foo' => 'bar' }, kind: :consumer }
+
+  # instrument instance function self.create_session
+  def self.create_session(user)
+  end
+
+  class << self
```

Comment:
```suggestion
  def create_session(user)
  end

  def self.create_session(user)
  end

  # instrument instance method create_session
  add_tracer :create_session, 'custom_name', { attributes: { 'foo' => 'bar' }, kind: :consumer }

  # instrument class method create_session
  class << self
```

Reply:
I think `create_session` is the instance method, while `self.create_session` is the class method. And is the `include SolarWindsAPM::API::Tracer` on line 93 needed by the "instrument instance method" usage? It might be clearer if we just show the instrument instance vs class method as two separate snippets.

Reply:
Do you mean like putting them in separate code block?

Reply:
> Also, how about the docstring

Updated

Reply:
Yes, i mean two separate code blocks, because currently it's unclear where `include SolarWindsAPM::API::Tracer` is needed for each scenario--the current example has it once at the start of class definition, then again via the `class << self` construct. If we keep a single code block example it would be good to clarify why.

Reply:
Updated